helpers/installcredprovider.ps1

```diff
@@ -111,13 +113,20 @@ function InstallZip {
         $releaseJson = $release.Content | ConvertFrom-Json
         $zipAsset = $releaseJson.assets | ? { $_.name -eq $zipFile }
         $packageSourceUrl = $zipAsset.browser_download_url
-    } catch {
-        Write-Error $zipErrorString
-        return
+                
+        if (!$releaseJson) {
+            throw("Unable to convert JSON")
+        }
+        if (!$zipAsset) {
+            throw("Unable to retrieve zip asset")
+        }
+        if (!$packageSourceUrl) {
+            throw("Unable to retrieve source URL")
+        }
     }
-
-    if (!$packageSourceUrl) {
-        Write-Error $zipErrorString
+    catch {
+        $zipErrorString = "$zipErrorString `nError : " + $_.Exception.Message 
```

Comment:
Since $zipErrorString is only used in one place anymore, you can delete it from line 109 and just have the error typed here to simplify the code. Would something like this work?
`Write-Error "Unable to resolve the browser download url from $releaseUrl Error: $_.Exception.Message"`